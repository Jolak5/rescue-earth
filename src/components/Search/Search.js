import Button from 'react-bootstrap/Button';
import Container from 'react-bootstrap/Container';
import Col from 'react-bootstrap/Col';
import { useDispatch, useSelector } from 'react-redux';
import { useState } from 'react';
import { clearFilteredState, filterItem, renderItem } from '../../redux/homepage/homepageSlice';
import { openModal } from '../../redux/modal/modalSlice';
import { closeSearch } from '../../redux/search/searchSlice';

// Render search
export default function Search() {
  const dispatch = useDispatch();
  const searchFilter = useSelector((state) => state.Forex.searchFilter);
  const [filter, setfilter] = useState('');
  function handleSubmit(e) {
    e.preventDefault();
    dispatch(filterItem(filter));
    setfilter('');
  }

  console.log(filter);

  // function handleChange(e) {
  //   setfilter(e.target.value);
  //   console.log(filter);
  // }
  return (
    <Container className="forex-container">
      <form onSubmit={handleSubmit}>
        <input type="text" id="kayode" value={filter} onChange={(e) => setfilter(e.target.value)} placeholder="country name" />
        <input type="submit" placeholder="submit" />
      </form>
      <button type="button" onClick={() => dispatch(closeSearch(), clearFilteredState())}>close</button>
      {searchFilter.map((item) => (
        <Col className="eachItem" key={item.id}>
          <h2>{item.country}</h2>
          <p>{item.death}</p>
          <Button onClick={() => {
            dispatch(renderItem(item.id));
            dispatch(openModal());
          }}
          >
            Kayode

          </Button>
        </Col>
      ))}
    </Container>
  );
}
