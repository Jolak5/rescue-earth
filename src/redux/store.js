import { configureStore } from '@reduxjs/toolkit';
import pollutionReducer from './homepage/homepageSlice';

export default configureStore({
  reducer: {
    pollution: pollutionReducer,
  },
});
