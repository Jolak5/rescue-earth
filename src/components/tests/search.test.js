import React from 'react';
import { BrowserRouter } from 'react-router-dom';
import { cleanup, render } from '@testing-library/react';
import { Provider } from 'react-redux';
import Search from '../Search/Search';
import store from '../../redux/store';

afterEach(cleanup);

describe('Navbar', () => {
  it('Search to match Snapshot', () => {
    const search = render(<BrowserRouter><Provider store={store}><Search /></Provider></BrowserRouter>);
    expect(search).toMatchSnapshot();
  });
});
