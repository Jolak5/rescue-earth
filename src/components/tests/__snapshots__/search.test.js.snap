// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Search to match Snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container"
      >
        <header
          class="search-header"
        >
          <a
            aria-current="page"
            class="active"
            href="/"
          >
            <svg
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
              />
            </svg>
          </a>
          <h1>
            CoviLab 
          </h1>
          <form>
            <input
              id="kayode"
              placeholder="country name"
              type="text"
              value=""
            />
            <input
              class="submit"
              placeholder="submit"
              type="submit"
            />
          </form>
        </header>
        <div
          class="global"
        >
          <div
            class="global-details stats"
          />
        </div>
        <div
          class="forex-container"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container"
    >
      <header
        class="search-header"
      >
        <a
          aria-current="page"
          class="active"
          href="/"
        >
          <svg
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M9.4 233.4c-12.5 12.5-12.5 32.8 0 45.3l160 160c12.5 12.5 32.8 12.5 45.3 0s12.5-32.8 0-45.3L109.2 288 416 288c17.7 0 32-14.3 32-32s-14.3-32-32-32l-306.7 0L214.6 118.6c12.5-12.5 12.5-32.8 0-45.3s-32.8-12.5-45.3 0l-160 160z"
            />
          </svg>
        </a>
        <h1>
          CoviLab 
        </h1>
        <form>
          <input
            id="kayode"
            placeholder="country name"
            type="text"
            value=""
          />
          <input
            class="submit"
            placeholder="submit"
            type="submit"
          />
        </form>
      </header>
      <div
        class="global"
      >
        <div
          class="global-details stats"
        />
      </div>
      <div
        class="forex-container"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
